### ISSUE TITLE
Klarna issue with free products

### ISSUE THREAD

**Issue by JoostWan** [buckaroo-it/Magento2#39]
We have a client that adds a free product to the shopping cart. The price for this product is zero. When the customer wants to place a order with Klarna we get a validation error. I did some debugging. When the price is zero with no tax percent the module sends an empty request for that article. See request below for GroupID 4. Thats the free product.

```
                                    [43] => Array
                                        (
                                            [_] =>
                                            [Group] => Article
                                            [GroupID] => 4
                                            [Name] => ArticlePrice
                                        )

                                    [44] => Array
                                        (
                                            [_] => 1
                                            [Group] => Article
                                            [GroupID] => 4
                                            [Name] => ArticleQuantity
                                        )

                                    [45] => Array
                                        (
                                            [_] =>
                                            [Group] => Article
                                            [GroupID] => 4
                                            [Name] => ArticleVat
                                        )

                                    [46] => Array
                                        (
                                            [_] => 2
                                            [Group] => Article
                                            [GroupID] => 5
                                            [Name] => ArticleNumber
                                        )

                                    [47] => Array
                                        (
                                            [_] => 4.89
                                            [Group] => Article
                                            [GroupID] => 5
                                            [Name] => ArticlePrice
                                        )
```

For now we changed the code on this lines. I don't know if its good but it's working on our side for now. 

https://github.com/buckaroo-it/Magento2/blob/master/Model/Method/Klarnakp.php#L953 

with `'_' => $item->getTaxPercent() ?? 0,`

and 

https://github.com/buckaroo-it/Magento2/blob/master/Model/Method/Klarnakp.php#L1008

with 

```
 public function calculateProductPrice($productItem, $includesTax)
    {
        if ($includesTax) {
            $productPrice = $productItem->getRowTotalInclTax() ?? 0;
        } else {
            $productPrice = $productItem->getRowTotal() ?? 0;
        }

        return $productPrice;
    }

```

**Comment by vladislav-padalka-hys** [comment]
Thanks for your report !
It was added to the master branch : https://github.com/buckaroo-it/Magento2/commit/7ddb432bd1006c8a898946eb86529f86f50c26de

**Comment by Buckaroo-Rens** [comment]
Fix included in latest release 1.25.2
